transmission.go

```diff
@@ -316,30 +257,60 @@ func (b *batch) Fire(notifier muster.Notifier) {
 					})
 				}
 			}
+			continue
 		}
-		return
-	}
 
-	for datasetName, resps := range batchResponse {
-		if events, ok := b.events[datasetName]; ok { // just in case
-			// if an Event triggered a JSON error, it wasn't sent to the server and
-			// won't have a returned response... so we have ot be a bit more careful
-			// matching up responses with Events.
-			var eIdx int
-			for _, resp := range resps {
-				resp.Duration = dur / time.Duration(b.numEncoded)
-				for events[eIdx] == nil {
-					eIdx++
-				}
-				if eIdx == len(events) { // just in case
-					break
-				}
-				resp.Metadata = events[eIdx].Metadata
-				b.enqueueResponse(resp)
+		// Go through the responses and send them down the queue. If an Event
+		// triggered a JSON error, it wasn't sent to the server and won't have a
+		// returned response... so we have to be a bit more careful matching up
```

Comment:
s/to be a bit more careful matching/to check for `nil`-ed out events while matching/